src/test/java/com/eleks/academy/whoami/repository/GameInMemoryRepositoryTest.java

```diff
@@ -3,47 +3,39 @@
 import com.eleks.academy.whoami.core.SynchronousGame;
 import com.eleks.academy.whoami.core.impl.PersistentGame;
 import com.eleks.academy.whoami.model.request.NewGameRequest;
+import com.eleks.academy.whoami.repository.impl.GameInMemoryRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
-
-import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
 @ExtendWith(MockitoExtension.class)
 public class GameInMemoryRepositoryTest {
 
-	@Mock
-	private final Map<String, SynchronousGame> games = new ConcurrentHashMap<>();
-
 	private final NewGameRequest gameRequest = new NewGameRequest();
 
+	@Mock
+	private final GameInMemoryRepository gameRepository = new GameInMemoryRepository();
+
 	@BeforeEach
 	public void setMockMvc() {
-		gameRequest.setMaxPlayers(5);
+		gameRequest.setMaxPlayers(4);
 	}
 
 	@Test
 	void save() {
 		final String player = "player";
 		SynchronousGame game = new PersistentGame(player, gameRequest.getMaxPlayers());
 
-		when(games.put(any(), any())).thenReturn(game);
-
-		var savedGame = this.games.put(game.getId(), game);
-
-		assertNotNull(savedGame);
-		assertNotNull(savedGame.getId());
-		assertNotNull(savedGame.getStatus());
-		assertNotNull(savedGame.getPlayersInGame());
+		assertThat(gameRepository.save(game))
+				.isEqualTo(gameRepository.save(new PersistentGame("player", 4)));
```

Comment:
just `.isEqualTo(new PersistentGame("player", 4))` you need to compare actual result with expected, but not 2 actual, if you do so you tests do not have sense

Reply:
Resolved